contracts/staking/Staking.sol

```diff
@@ -372,59 +337,289 @@ contract Staking is
         onlyDefinedStakingRate(token, chainId)        
         returns(uint tokenAmount)
     {
-        IInstanceDataProvider.TokenInfo memory info = _bundleRegistry.getTokenInfo(token, chainId);
+        IInstanceDataProvider.TokenInfo memory info = _registry.getTokenInfo(token, chainId);
         uint256 rate = _stakingRate[token][chainId];
         return (_math.mul(dipAmount, rate) * 10 ** info.decimals) / 10 ** DIP_DECIMALS;
     }
 
-    function getRewardRate() 
+
+    function targets() external override view returns(uint256 numberOfTargets) {
+        return _targetIds.length;
+    }
+
+
+    function getTargetId(uint256 idx) external override view returns(bytes32 targetId) {
+        require(idx < _targetIds.length, "ERROR:STK-200:TARGET_INDEX_TOO_LARGE");
+        return _targetIds[idx];
+    }
+
+
+    function isTarget(bytes32 targetId) external override view returns(bool isATarget) {
+        return _target[targetId].targetType != TargetType.Undefined;
+    }
+
+    function isTargetRegistered(Target memory target) 
+        external override 
+        view 
+        returns(bool isRegistered)
+    {
+        if(target.targetType == TargetType.Instance) {
+            return _registry.isRegisteredInstance(target.instanceId);
+        } else if(target.targetType == TargetType.Component) {
+            return _registry.isRegisteredComponent(target.instanceId, target.componentId);
+        } else if(target.targetType == TargetType.Bundle) {
+            return _registry.isRegisteredBundle(target.instanceId, target.bundleId);
+        } else {
```

Comment:
What about `protocol` target? this would create a `false` response.

Reply:
correct for now, there is no way to register the protocol for now...